crates/pink/sidevm/sidevm/src/channel.rs

```diff
@@ -81,6 +83,24 @@ impl Future for Next<'_, Message> {
     }
 }
 
+impl Future for Next<'_, SystemMessage> {
+    type Output = Option<SystemMessage>;
+
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        let waker_id = crate::env::tasks::intern_waker(cx.waker().clone());
+        match ocall::poll(waker_id, self.ch.res_id.0) {
+            Ok(msg) => {
+                let message =
+                    SystemMessage::decode(&mut &msg[..]).expect("Failed to decode SystemMessage");
```

Comment:
This way, if we introduce a new `SystemMessage`, the sidevm will crash

Reply:
Fixed